jetty-core/jetty-tests/jetty-test-client-transports/src/test/java/org/eclipse/jetty/test/client/transport/HttpClientStreamTest.java

```diff
@@ -298,17 +298,10 @@ public void process(Request request, org.eclipse.jetty.server.Response response,
         InputStreamResponseListener listener = new InputStreamResponseListener()
         {
             @Override
-            public void onContent(Response response, ByteBuffer content, Callback callback)
+            public void onContent(Response response, Content.Chunk chunk, Runnable demander)
             {
-                super.onContent(response, content, new Callback()
-                {
-                    @Override
-                    public void failed(Throwable x)
-                    {
-                        latch.countDown();
-                        callback.failed(x);
-                    }
-                });
+                latch.countDown();
+                super.onContent(response, chunk, demander);
```

Comment:
The latch is counted down too early, so it's in a race with the asserts at the end of the method.
I think the 2 lines above should be swapped.

Reply:
You're right. Let me fix this.